Author: Steve Langasek <steve.langasek@ubuntu.com>
Description: Fix build failure on ppc64el
 On ppc64/ppc64el, the Linux kernel headers default to defining __u64 as
 'unsigned long int' for "compatibility".  However, this is incompatible
 with android-tools' own definition of the type.  Use the
 __SANE_USERSPACE_TYPES__ define to get kernel header behavior consistent
 with other architectures.
---
 system/extras/ext4_utils/ext4_utils.c       |    1 +
 system/extras/ext4_utils/make_ext4fs_main.c |    3 ++-
 system/extras/ext4_utils/wipe.c             |    1 +
 3 files changed, 4 insertions(+), 1 deletion(-)

--- a/system/extras/ext4_utils/ext4_utils.c
+++ b/system/extras/ext4_utils/ext4_utils.c
@@ -37,6 +37,7 @@
 #endif
 
 #if defined(__linux__)
+#define __SANE_USERSPACE_TYPES__
 #include <linux/fs.h>
 #elif defined(__APPLE__) && defined(__MACH__)
 #include <sys/disk.h>
--- a/system/extras/ext4_utils/make_ext4fs_main.c
+++ b/system/extras/ext4_utils/make_ext4fs_main.c
@@ -20,6 +20,7 @@
 #include <unistd.h>
 
 #if defined(__linux__)
+#define __SANE_USERSPACE_TYPES__
 #include <linux/fs.h>
 #elif defined(__APPLE__) && defined(__MACH__)
 #include <sys/disk.h>
@@ -183,7 +184,7 @@ int main(int argc, char **argv)
 		}
 		fs_config_func = canned_fs_config;
 	} else if (mountpoint) {
-		fs_config_func = fs_config;
+    fs_config_func = fs_config;
 	}
 
 	if (wipe && sparse) {
--- a/system/extras/ext4_utils/wipe.c
+++ b/system/extras/ext4_utils/wipe.c
@@ -21,6 +21,7 @@
 
 #if defined(__linux__)
 
+#define __SANE_USERSPACE_TYPES__
 #include <linux/fs.h>
 #include <sys/ioctl.h>
 
